docker-compose.yml

```diff
@@ -1,77 +1,106 @@
-version: '3.4'
+# Any value can be overridden in docker-compose.override.yml, a file that is not
+# under VCS control. See docker-compose.override.yml.dist for more details.
+version: '3'
 services:
   web:
-    build:
-      context: ./resources/docker/web
+    image: fpfis/httpd-php-dev:7.1
+    working_dir: /var/www/html
     ports:
-      - "8080:80"
+      - '8080:8080'
+    # macOS users should override 'volumes:' in docker-compose.override.yml.
+    # See docker-compose.override.yml.dist.
     volumes:
-      - .:/var/www/html
-      - ./build.docker.main.xml:/var/www/html/build.xml
+      - ${PWD}:/var/www/html
     env_file:
       - .env.dist
     environment:
       DOCUMENT_ROOT: "/var/www/html/web"
-      PHP_XDEBUG_ENABLED: "1"
-      PHP_IDE_CONFIG: "serverName=Docker"
-      PHP_SENDMAIL_PATH: "/usr/sbin/sendmail -t -i -S mail:1025"
+#      PHP_XDEBUG_ENABLED: "0"
+#      PHP_IDE_CONFIG: "serverName=Docker"
+#      PHP_SENDMAIL_PATH: "/usr/sbin/sendmail -t -i -S mail:1025"
+      SMTP_SERVER: mail
+      SMTP_PORT: 1025
+      SMTP_FROM: contact@example.com
+      # Environment variables without a value will receive the values from the
+      # host machine, if they are set.
+      # See https://docs.docker.com/compose/environment-variables/#pass-environment-variables-to-containers
+      DRUPAL_HASH_SALT:
+      ASDA_URL:
+      ASDA_USER:
+      ASDA_PASSWORD:
+    depends_on:
+      - mysql
+      - virtuoso
+      - solr
+      - redis
+      - mail
+      - selenium
 
   mysql:
-    image: mysql:5.6
+    image: percona/percona-server:5.7
+    # macOS users should override 'volumes:' in docker-compose.override.yml.
+    # See docker-compose.override.yml.dist.
+    volumes:
+      - ./resources/docker/mysql/config:/etc/mysql/conf.d
+    ports:
+      # Avoid port collision with a potential host machine instance of MySQL.
+      - '3307:3306'
     environment:
-      MYSQL_ALLOW_EMPTY_PASSWORD: "no"
-      MYSQL_USER: 'db_user'
-      MYSQL_PASSWORD: 'db_pass'
-      MYSQL_DATABASE: 'db_name'
+      MYSQL_ALLOW_EMPTY_PASSWORD: "yes"
 
   virtuoso:
     image: tenforce/virtuoso
     environment:
       - SPARQL_UPDATE=true
-      - DBA_PASSWORD=dba
+      # Virtuoso .ini overrides. Pattern: VIRT_$SECTION_$KEY=$VALUE
+      # See https://hub.docker.com/r/tenforce/virtuoso
+      - VIRT_Parameters_O_DIRECT=1
+      - VIRT_Parameters_ThreadCleanupInterval=1
+      - VIRT_Parameters_NumberOfBuffers=340000
+      - VIRT_Parameters_MaxDirtyBuffers=250000
+      - VIRT_Replication_ServerEnable=0
+      - VIRT_SPARQL_ResultSetMaxRows=1000000
+    ports:
+      # Avoid port collision with a potential host machine instance of Virtuoso.
```

Comment:
I'm using local development but I want to be able to run the composer version in parallel. I preferred the "safe way" by default. I don't think `docker-compose.yml` being too big is a real issue, it's 2 lines for each service. Also it could be a good sample to customize with a different port. But if you think this is a real problem I can delete it.